Dataset.py

```diff
@@ -259,6 +259,20 @@ def get_seq_length(self, seq_idx):
     """
     raise NotImplementedError
 
+  def get_estimated_seq_length(self, seq_idx):
+    """
+    In contrast to self.get_seq_length(), this method is designed to work for sequences that have not been loaded yet
+    via self.load_seqs().
+    Used by meta-datasets for sequence ordering. Currently we only provide one number, i.e. do not give different
+    estimates for the different data keys (as in get_seq_length()). It is up to the dataset what this number represents
+    and how it is computed.
+
+    :param int seq_idx:
```

Comment:
You should clarify this `seq_idx` further. I think you even have confused this yourself (see my other comment below).
* There is the seq idx related to the current epoch (after you called `init_seq_order`) (this is the same seq idx you would use for `load_seqs` or `get_data`).
* There is a seq idx globally for the dataset (independent of sorting, filtering, etc). In other contexts, I call this "real seq idx", or "corpus seq idx".
This real seq idx might not be well defined for all datasets.
If the function `get_corpus_seq_idx` is defined, it will get epoch seq idx -> corpus seq idx.
If `get_all_tags` is defined, the corpus seq idx is an index in this list.

You should make clear which of these two is expected here.
As I understood your reasoning, you want to have this actually considering the current epoch seq order. So it would use the seq idx related to the current epoch. And not the corpus seq idx.
It is enough if you just write:
```suggestion
    :param int seq_idx: for current epoch, not the corpus seq idx
```